cmd/skaffold/app/cmd/init.go

```diff
@@ -63,13 +65,23 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 	cmd.Flags().StringVarP(&opts.ConfigurationFile, "filename", "f", "skaffold.yaml", "Filename or URL to the pipeline file")
 	cmd.Flags().BoolVar(&skipBuild, "skip-build", false, "Skip generating build artifacts in skaffold config")
 	cmd.Flags().BoolVar(&force, "force", false, "Force the generation of the skaffold config")
+	cmd.Flags().StringVar(&composeFile, "compose-file", "", "Initialize from a docker-compose file")
 	cmd.Flags().StringArrayVarP(&cliArtifacts, "artifact", "a", nil, "'='-delimited dockerfile/image pair to generate build artifact\n(example: --artifact=/web/Dockerfile.web=gcr.io/web-project/image)")
 	return cmd
 }
 
 func doInit(out io.Writer) error {
 	rootDir := "."
 
+	if composeFile != "" {
+		// run kompose first to generate k8s manifests, then run skaffold init
+		logrus.Infof("running 'kompose convert' for file %s", composeFile)
+		komposeCmd := exec.Command("kompose", "convert", "-f", composeFile)
```

Comment:
do we have a preference around supported version for `kompose`?

Reply:
just added it to the dockerfile with a pinned version. I only tried with the latest (v1.17.0), but I'm assuming any older version that does what it's supposed to do will work with skaffold.